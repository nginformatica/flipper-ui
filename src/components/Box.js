import React from 'react'
import _ from 'prop-types'
import styled from 'styled-components'
import { white } from '../colors'

const StyledBox = styled.div`
    background-color: ${white};
    font-family: 'Roboto', sans-serif;
    line-height: 1.2em;
    opacity: 0.95;
    border-radius: 6px;
    min-height: 400px;
    margin-bottom: 10px;
    padding: 28px;
`

const Box = ({ children, ...otherProps }) =>
    <StyledBox { ...otherProps }>
        { children }
    </StyledBox>

Box.propTypes = {
    style: _.object,
    primary: _.bool,
    children: _.node
}

export default Box
