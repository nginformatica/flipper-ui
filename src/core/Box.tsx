import React, { FC } from 'react'
import type { DefaultProps } from './types'
import Paper from './Paper'

interface BoxProps extends DefaultProps {
    minHeight?: number
}

const Box: FC<BoxProps> = ({
    children,
    margin,
    padding = 18,
    style = {},
    minHeight = 400,
    ...otherProps
}) =>
    <Paper
        padding={ padding }
        margin={ margin }
        style={ { minHeight, ...style } }
        { ...otherProps }>
        { children }
    </Paper>

export default Box
