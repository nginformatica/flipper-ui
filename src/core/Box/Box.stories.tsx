import React from 'react'
import { ComponentStory, ComponentMeta } from '@storybook/react'
import Box from '.'

export default {
    title: 'Box',
    component: Box
} as ComponentMeta<typeof Box>

const Template: ComponentStory<typeof Box> = args => <Box {...args} />

export const Default = Template.bind({})

Default.args = {
    margin: 25,
    children: 'Open the box and think outside of it.'
}
