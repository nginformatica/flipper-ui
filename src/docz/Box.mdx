---
name: Box
route: /Box
---

import Box from '../components/Box'
import { Playground, PropsTable } from 'docz'

# Box

## Properties
<PropsTable of={ Box } />

## Demo
<Playground style={ { padding: '10px', backgroundColor: 'silver' } }>
    <Box>
        Foi assim;
        No dia em que todas as pessoas;
        Do planeta inteiro;
        Resolveram que ninguém ia sair de casa;
        Como que se fosse combinado em todo;
        O planeta;
        Naquele dia, ninguém saiu de casa, ninguém ninguém;
        O empregado não saiu pro seu trabalho;
        Pois sabia que o patrão também não tava lá;
        Dona de casa não saiu pra comprar pão;
        Pois sabia que o padeiro também não tava lá;
        E o guarda não saiu para prender;
        Pois sabia que o ladrão, também não tava lá;
        E o ladrão não saiu para roubar;
    </Box>
</Playground>
