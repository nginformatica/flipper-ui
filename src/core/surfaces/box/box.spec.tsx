import React from 'react'
import { render, screen } from '@testing-library/react'
import Box from '.'

describe('Advertise', () => {
    it('should render', () => {
        render(
            <Box data-testid='box-container'>
                <div>Box</div>
            </Box>
        )

        const text = screen.getByText('Box')
        const container = screen.getByTestId('box-container')

        expect(text).toBeDefined()
        expect(container).toHaveProperty('style.padding', '18px')
        expect(container).toHaveProperty('style.minHeight', '400px')
    })

    it('should render with custom string margins', () => {
        render(
            <Box margin='10px' padding='12px' data-testid='box-container'>
                <div>Box</div>
            </Box>
        )

        const container = screen.getByTestId('box-container')

        expect(container).toHaveProperty('style.margin', '10px')
        expect(container).toHaveProperty('style.padding', '12px')
    })

    it('should render with custom number margins', () => {
        render(
            <Box margin={10} padding={12} data-testid='box-container'>
                <div>Box</div>
            </Box>
        )

        const container = screen.getByTestId('box-container')

        expect(container).toHaveProperty('style.margin', '10px')
        expect(container).toHaveProperty('style.padding', '12px')
    })

    it('should match snapshot', () => {
        const { container } = render(
            <Box margin={10} padding={12} data-testid='box-container'>
                <div>Box</div>
            </Box>
        )

        expect(container).toMatchSnapshot()
    })
})
