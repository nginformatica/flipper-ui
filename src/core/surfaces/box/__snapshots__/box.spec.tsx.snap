// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Advertise should match snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
    data-testid="box-container"
    role="mui-paper-container"
    style="padding: 12px; margin: 10px; min-height: 400px;"
  >
    <p
      class="MuiTypography-root MuiTypography-body2"
    >
      Box
    </p>
  </div>
</div>
`;
